Adding NUCLEO_WL55JC support

JeanMarcR opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

We would like to introduce the new STM32WL family.

Complementing the STM32 RF connectivity portfolio, the STM32WL System-On-Chip integrates both a general purpose microcontroller and a sub-GHz radio on the same chip.
Built on Arm® Cortex®‐M4 and Cortex®-M0+ cores (single- and dual-core architectures available - 48 MHz), STM32WL microcontrollers support multiple modulations– LoRa®, (G)FSK, (G)MSK, BPSK – to ensure flexibility in wireless applications with LoRaWAN®, Sigfox, W-MBUS or any other suitable protocol in a fully open way.
Thanks to a deep integration, the innovative and open architecture is optimized for LoRaWAN® legacy, Sigfox, proprietary protocols, flexible resource use, power management and helps lower BOM cost while offering a better user experience.

Developed using the same technology as the one implemented in our ultra-low-power STM32L4 microcontrollers, the STM32WL series provides similar digital and analog peripherals for basic or complex application use cases requiring an extended battery life and a long RF range through its sub-GHz transceiver.

To ensure worldwide compatibility, the STM32WL MCUs feature a dual power output and a wide linear frequency range fitting any unlicensed RF spectrum need.

Overall, the STM32WL series is the STM32 family’s pioneer in sub-GHz wireless connectivity, offering ease-of-use and reliability, while being perfectly tailored for a wide range of industrial and consumer applications.

Cube FW files are now available:
https://www.st.com/en/embedded-software/stm32cubewl.html

MBED-OS port has been implemented for NUCLEO_WL55JC target:

256 Kbytes of Flash memory
64 Kbytes of SRAM

https://www.st.com/en/microcontrollers-microprocessors/stm32wl55jc.html

 
### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Target update
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    Test executed on STMicroelectronics test bench with toolchain: ARM


    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

facchinm committed:
BlockDevices: specify mbed namespace where needed

taunomagnusson committed:
Update README.md

taunomagnusson committed:
Update README.md

Commit:
Added entry for STM32 NUCLEO_G431KB Board

Commit:
Adding support for NUCLEO_G431KB. Created this file based on G474RE. Fully verified with Spec Sheets

Commit:
Adding support for NUCLEO_G431KB. Created this file based on G474RE. I have done a decent effort to port it / check it. However - Warning - Not everything is verified the the G431 Spec Sheets.

Commit:
Adding support for NUCLEO_G431KB. Created this file based on G474RE file. Added support for USB - connected the HSI48 clock to USB (and RNG). Warning: I'm assuming Flash latency is the same as for G474RE.

Commit:
Fix: Added ALIGN(8) for __etext section. This existed in mbed-6.3.0 version. I have not verified if it is truly needed, but it is better to err on the safe side.

Commit:
G431 only has 1 Flash Bank. Code is written with the assumption all G4 devices has 2 Flash Banks. Fix: Commented out SYSCFG_MEMRMP_FB_MODE since stm32g431KB only has 1 Flash Bank. flash_api.c checks for this #define and if it's set understands it as if the chip has 2 flash banks. STM32G4xx_HAL_Driver/stm32g4xx_hal.c also needs to check for this #define since it has flash bank swapping commands that are called in the startup sequence. Fix: Check for SYSCFG_MEMRMP_FB_MODE in Enable/Disable Flash bank swapping.

Commit:
Add: Added STM32G4 series to LP USB list

Commit:
Reverted to original README.md file

Commit:
Adding CMakeLists.txt files for NUCLEO_G431KB

Commit:
Updated G431KB inline with review comments

Commit:
Cleaned up and verified file

Commit:
fix namespaces for MBED_NO_GLOBAL_USING_DIRECTIVE feature

added namespaces in various files

hugueskamba committed:
NFC: Add support for M24SR driver

The repository https://github.com/ARMmbed/mbed-nfc-m24sr
is no longer required to build applications that require
this driver.

hugueskamba committed:
CMake: Add support for Mbed TLS Greentea tests

The Mbed TLS Greentea tests can now be built with CMake.

jeromecoutant commented on tools/export/uvision/__init__.py:
```diff
@@ -171,7 +171,7 @@ def format_flags(self):
             ",".join("-D{}".format(s) for s in
                      self.toolchain.get_symbols(for_asm=True)))
-        flags['asm_flags'] = asm_flag_string
+        flags['asm_flags'] = "{} {}".format(" ".join(self.toolchain.flags['asm']), asm_flag_string)
```
remove this patch from PR => followed by #10424

JeanMarcR replied:
Done

jeromecoutant commented on tools/cmake/cores/Cortex-M4.cmake:
```diff
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Sets cpu core options
+
```
remove patch from this PR => #14185

JeanMarcR replied:
Done

jeromecoutant commented on targets/targets.json:
```diff
@@ -4076,6 +4076,67 @@
+            "ANALOGOUT",
+            "TRNG"
+        ],
+        "device_has_remove": [
+        ]
```
to remove

JeanMarcR replied:
Done

jeromecoutant commented on targets/targets.json:
```diff
@@ -4076,6 +4076,67 @@
+            "CRC",
+            "FLASH",
+            "SERIAL_ASYNCH",
+            "ANALOGOUT",
```
Miss ANALOGIN ?

JeanMarcR replied:
Yes. I am updating.

jeromecoutant commented on targets/TARGET_STM/trng_api.c:
```diff
@@ -37,7 +37,7 @@ void trng_init(trng_t *obj)
     uint32_t dummy;
 
-#if defined(RCC_PERIPHCLK_RNG) /* STM32L4 / STM32H7 / STM32WB / STM32G4 */
+#if defined(RCC_PERIPHCLK_RNG) /* STM32L4 / STM32H7 / STM32WB */ /* STM32WL*/
```
keep STM32G4 comment

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/trng_api.c:
```diff
@@ -48,14 +48,19 @@ void trng_init(trng_t *obj)
     PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_RNG;
     PeriphClkInitStruct.RngClockSelection = RCC_RNGCLKSOURCE_PLL;
-#if defined(DUAL_CORE)
+#elif defined(TARGET_STM32WL)
+    RCC_PeriphCLKInitTypeDef PeriphClkInitStruct;
+    /*Select PLLQ output as RNG clock source*/
+    PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_RNG;
+    PeriphClkInitStruct.RngClockSelection = RCC_RNGCLKSOURCE_MSI;
```
This should be before #elif defined(TARGET_STM32H7)
because next #if defined(DUAL_CORE) has no sense with WL...

JeanMarcR replied:
Do not take into account previous comment.
Done.

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/spi_device.h:
```diff
@@ -0,0 +1,27 @@
+#include "stm32wlxx_ll_spi.h"
+
+// Defines the word legnth capability of the device where Nth bit allows for N window size
+//#define STM32_SPI_CAPABILITY_WORD_LENGTH (0x00008080)
```
remove this line

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/spi_device.h:
```diff
@@ -0,0 +1,27 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/spi_api.c:
```diff
@@ -0,0 +1,59 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/pwmout_device.h:
```diff
@@ -0,0 +1,49 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

jeromecoutant replied:
ping

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/pwmout_device.c:
```diff
@@ -0,0 +1,40 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/objects.h:
```diff
@@ -0,0 +1,158 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/i2c_device.h:
```diff
@@ -0,0 +1,82 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/flash_api.c:
```diff
@@ -0,0 +1,229 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * Copyright (c) 2019 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/cmsis.h:
```diff
@@ -0,0 +1,24 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 STMicroelectronics.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/analogin_device.c:
```diff
@@ -0,0 +1,197 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WLE5xC/TOOLCHAIN_ARM/stm32wle5xc.sct:
```diff
@@ -0,0 +1,57 @@
+#! armcc -E
```
As you know, to align with other targets

JeanMarcR replied:
Done

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WL55xC/TOOLCHAIN_ARM/stm32wl55xc.sct:
```diff
@@ -0,0 +1,57 @@
+#!armclang -E --target=arm-arm-none-eabi -mcpu=cortex-m7 -xc
```
not M7

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WL55xC/TARGET_NUCLEO_WL55JC/system_clock.c:
```diff
@@ -0,0 +1,93 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * Copyright (c) 2019 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the CFG_HW_RCC_SEMID at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
align license header with us_ticker_data.h for ex

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WL55xC/TARGET_NUCLEO_WL55JC/system_clock.c:
```diff
@@ -0,0 +1,93 @@
+  *-----------------------------------------------------------------------------
+  * System clock source | HSE (external 32 MHz clock)
+  *-----------------------------------------------------------------------------
+  * SYSCLK(MHz)         | 32
+  * AHBCLK (MHz)        | 32
+  * APB1CLK (MHz)       | 32
+  * APB2CLK (MHz)       | 32
```
Write correct values : 48?

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WL55xC/TARGET_NUCLEO_WL55JC/system_clock.c:
```diff
@@ -0,0 +1,93 @@
+
+#include "stm32wlxx.h"
+#include "mbed_error.h"
+#include "stm32wlxx_ll_hsem.h"
```
not used ?

JeanMarcR replied:
Indeed not used. Updated.

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WL55xC/TARGET_NUCLEO_WL55JC/system_clock.c:
```diff
@@ -0,0 +1,93 @@
+#include "stm32wlxx_ll_hsem.h"
+#include "objects.h"
+
+void Error_Handler(void)
+{
+  /* USER CODE BEGIN Error_Handler_Debug */
+  /* User can add his own implementation to report the HAL error return state */
+  while (1)
+  {
+  }
+  /* USER CODE END Error_Handler_Debug */
+}
```
to remove

JeanMarcR replied:
Done

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WL55xC/TARGET_NUCLEO_WL55JC/system_clock.c:
```diff
@@ -0,0 +1,93 @@
+    Error_Handler();
+  }
+
+  /* Peripheral clock enable */
+  /* __HAL_RCC_RNG_CLK_ENABLE(); A DECOMMENTER PEUT-ETRE */
```
to remove

JeanMarcR replied:
Done.

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WL55xC/TARGET_NUCLEO_WL55JC/system_clock.c:
```diff
@@ -0,0 +1,93 @@
+
+  if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_2) != HAL_OK)
+  {
+    Error_Handler();
```
use MBED_ASSERT

JeanMarcR replied:
Done.

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WL55xC/TARGET_NUCLEO_WL55JC/system_clock.c:
```diff
@@ -0,0 +1,93 @@
+  * @retval None
+  */
+
+void SetSysClock(void)
```
add MBED_WEAK to allow application to reconfigure clock

JeanMarcR replied:
Done.

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32WL/STM32Cube_FW/CMSIS/stm32wl55xx.h:
```diff
@@ -0,0 +1,11449 @@
+#define __VTOR_PRESENT            1U /*!< Vector Table Register supported               */
+#define __NVIC_PRIO_BITS          4U /*!< STM32WLxx uses 4 Bits for the Priority Levels */
+#define __Vendor_SysTickConfig    0U /*!< Set to 1 if different SysTick Config is used  */
+#define __FPU_PRESENT             0U /*!< FPU present                                   */
```
To remove warning:
```suggestion
#define __FPU_PRESENT             0 /*!< FPU present                                   */
```

jeromecoutant replied:
forget this comment, keep original file

JeanMarcR replied:
OK

hugueskamba commented on targets/TARGET_STM/TARGET_STM32WL/TARGET_STM32WLE4x8/TOOLCHAIN_ARM/stm32wle4x8.sct:
```diff
@@ -0,0 +1,57 @@
+#! armcc -E
```
```suggestion
#! armclang -E --target=arm-arm-none-eabi -x c -mcpu=cortex-<CORE_TYPE>
```

Please make sure the shebang in all scatter files specify `armclang` rather than `armcc`.🙌

jeromecoutant replied:
Ack
@JeanMarcR see patch on our internal branch

JeanMarcR replied:
Done

Commit:
Fixed a typo (incorrectly referring to G474RE instead of G431KB)

Commit:
Cleaning out workaround patches for G431KB now that Flash Bank Switching has been fixed on G4xxx level (ARMmbed/mbed-os patch #14149)

Commit:
USBDEVICE removed from G431KB as per ARMmbed policy (only boards with physical USBDEVICE should define it in targets.json)

Commit:
Reverted to a pure copy of 474RE system_clock.c since it was updated with RNG clock fixes. Removing USB Clock enablement (USB will not work out-of-the box but will require users to modify system_clock.c to enable its clock) - this inline with changes to targets.json where I removed the USBDEVICE define

Commit:
Autogenerated file using 'python targets/TARGET_STM/tools/STM32_gen_PeripheralPins.py -t M40_Nucleo_NUCLEO-G431KB'. Had to do some modifications since the autogenerated files contained errors.

0xc0170 committed:
CMake cores: fix NS cores

Set DOMAIN_NS=1 only for _NS cores. We had it otherway around.
I verified one non NS target that was failing to build, plus also checked the old tools
that had this definitions:

```
   91:     "Cortex-M23-NS":   ["__CORTEX_M23", "ARM_MATH_ARMV8MBL", "DOMAIN_NS=1",
   92                          "__CMSIS_RTOS", "__MBED_CMSIS_RTOS_CM"],
   93      "Cortex-M23":      ["__CORTEX_M23", "ARM_MATH_ARMV8MBL", "__CMSIS_RTOS",
   94                          "__MBED_CMSIS_RTOS_CM"],
```

jeromecoutant commented on targets/TARGET_STM/gpio_irq_api.c:
```diff
@@ -323,7 +323,7 @@ int gpio_irq_init(gpio_irq_t *obj, PinName pin, gpio_irq_handler handler, uint32
     /* Enable SYSCFG Clock */
-#if !defined(TARGET_STM32WB)
+#if (!defined(TARGET_STM32WB) && !defined(TARGET_STM32WL))
     __HAL_RCC_SYSCFG_CLK_ENABLE();
```
In this file, there is 
if defined(DUAL_CORE)
without && (H7) ==> to be added

Good catch @ludoch-stm

JeanMarcR replied:
Done.

Commit:
Fixed whitespaces (a tab that should be spaces)

Commit:
Updated to enable HSI48 Clock also for USBDEVICE define in targets.json. Also added so that the CK48 Clock Mux selects HSI48 for both RNG and USB. Without these the RNG/USB will be clocked at PLLQ (170 MHz) and not work.

Commit:
Removed the HAL_RCCEx_PeriphCLKConfig statement that was selecting the HSI48 CLK for USB. Assumption is that the STM32G4 reset value (defaults to HSI48 selected for USB) is preserved throughout Mbed initialization.

Commit:
After checking other parts of the code (trng_api.c and a few other board platforms (NUCLEO_F413ZH, ...)) it appears that even though the reset value defaults correctly it may be good practice to keep the HAL_RCCEx_PeriphCLKConfig statement that selects the HSI48CLK as source for USB/RNG.

JeanMarcR committed:
STM32WL : ADDING STM32 SUPPORT

Add code concerning all STM32 platforms

JeanMarcR committed:
STM32WL : ADDING STM32WL SUPPORT

Add code concerning all STM32WL platforms
- STM32 Cube Firmware

Update CMakeLists.txt due to mbedtools evolution

JeanMarcR committed:
STM32WL : ADDING STM32WL55xC SUPPORT

Add code concerning all STM32WL55xC platforms
- Scatter loader and start-up files for
  ARM, GCC and IAR compilers.
- cmsis file
- Update CMakeLists.txt due to mbtools evolution

JeanMarcR committed:
STM32WL : ADDING NUCLEO STM32WL55JC SUPPORT

Add code concerning all STM32WL55JC platforms
- system clock, pin and peripheral definition
  mbedtools make file

Modify CmakeList to adapt to mbedtools evolution

JeanMarcR committed:
STM32WL : ADDING TARGET SUPPORT

Add STM32WL information in targets.json file

JeanMarcR committed:
STM32WL : ADDING ARM PACK MANAGER SUPPORT

Add STM32WL information in index.json file
Adding sector

jeromecoutant committed:
LoraRadio test update for baremetal support

0xc0170 committed:
CMake cores: fix cpu flag

Couple of cores did not have common flag `--mcpu` set

paul-szczepanek-arm committed:
upgrade gtest to 1.10

0xc0170 committed:
CMake cores: remove march, not needed if we use mcpu

0xc0170 committed:
CMake cores: fix dsp for M33FE (enabling it)

I realized we specified nodsp for M33 core but had march for FE parts enabled with dsp.
This fixes it.

M33F has no dsp. M33FE cores have dsp enabled.

0xc0170 committed:
CMake cores: remove dsp as it is on by default

rwalton-arm committed:
cmake: Remove whitespace from mbed-os/CMakeLists.txt

rwalton-arm committed:
cmake: Refactor mbed_set_mbed_target_linker_script

Move the mbed_set_mbed_target_linker_script function into its own file
and make it set the linkerscript on the target passed in, rather than
setting the linkerscript on the top level mbed-core target. We use
CMake's automatic forwarding of usage requirements to forward the linker
script to the top level target.

This commit also adds a condition to check whether the MBED_TARGET is a
CMake buildsystem target. If it is we will just link to it in the
idiomatic CMake way, if it's not then we use the 'old' mechanism of
setting the linkerscript for the mbed-core target. This check will be
useful when we refactor MBED_TARGETS to be CMake buildsystem targets as
it will allow us to do the refactoring incrementally.

0xc0170 committed:
CMake cores: align fpu settings

Move the fpu settings across all cores so they are aligned.
Some cores did not have fpu set for ARMClang. I matched here both toolchains to set up
the fpu the same (Only exception is soft vs hard - that is separate issue I would not like to touch
in this series).

paul-szczepanek-arm committed:
update version in readme

JeanMarcR commented:
@jeromecoutant Please, could you review my push ? Thanks.

jeromecoutant commented on targets/TARGET_STM/trng_api.c:
```diff
@@ -51,14 +51,18 @@ void trng_init(trng_t *obj)
     if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInitStruct) != HAL_OK) {
         error("RNG clock configuration error\n");
     }
-#if defined(DUAL_CORE)
```
You removed H7 code...

JeanMarcR replied:
Yes because we have this line before
if defined(TARGET_STM32H7)

JeanMarcR replied:
Done

ladislas committed:
Update minimum version of intelhex to 2.3.0

paul-szczepanek-arm committed:
return OK when stopping advertising that is not active

kjansen-adi committed:
Adding USB HS Support to the SDP-K1

multiplemonomials committed:
Set executable suffix in CMake

multiplemonomials committed:
Remove CMAKE_EXECUTABLE_SUFFIX where it doesn't belong.

JeanMarcR commented:
@jeromecoutant
Please, could you review my push ?

rwalton-arm committed:
cmake: Refactor ST targets

Refactor all ST targets to be CMake buildsystem targets. This removes
the need for checking MBED_TARGET_LABELS repeatedly and allows us to be
more flexible in the way we include MBED_TARGET source in the build.

A side effect of this is it will allow us to support custom targets
without breaking the build for 'standard' targets, as we use CMake's
standard mechanism for adding build rules to the build system, rather
than implementing our own layer of logic to exclude files not needed for
the target being built. Using this approach, if an MBED_TARGET is not
linked to using `target_link_libraries` its source files will not be
added to the build. This means custom target source can be added to the
user's application CMakeLists.txt without polluting the build system
when trying to compile for a standard MBED_TARGET.

adbridge committed:
Expose the public Stream API

Stream.h contains a public API which is not currently exposed via
mbed.h (when it should be). This commit fixes this.

0xc0170 committed:
Add cortex-m55 to CMake support

0xc0170 committed:
CMake rtos: add M55 to irq handlers, use m33 handlers = the same

multiplemonomials committed:
Add newline

0xc0170 committed:
Merge pull request #14092 from hugueskamba/hk_nfc_m24sr_support

NFC: Add support for M24SR driver

0xc0170 committed:
Merge pull request #14197 from paul-szczepanek-arm/gtest

Upgrade gtest to 1.10

multiplemonomials committed:
Bring back CMAKE_EXECUTABLE_SUFFIX in map filename, with a vengeance!

0xc0170 committed:
Merge pull request #14201 from ladislas/ladislas-update-python-intelhex

Update maximum version of intelhex to 2.3.0

0xc0170 committed:
Merge pull request #14181 from hugueskamba/hk_cmake_greentea_mbedtls

CMake: Add support for Mbed TLS Greentea tests

0xc0170 committed:
Merge pull request #14196 from 0xc0170/fix-cores-gcc

CMake cores: fix cpu/fpu flags

rajkan01 committed:
CMake: rename greentea test macro

flippy84 committed:
STM: prevent task switch errors in HAL_GetTick

0xc0170 committed:
Merge pull request #14212 from adbridge/master

Expose the public Stream API

0xc0170 committed:
Merge pull request #14202 from paul-szczepanek-arm/fix-stop-advert

BLE: return OK when stopping advertising set that is not active

0xc0170 committed:
Merge pull request #14195 from jeromecoutant/PR_LORA_BARE

LoraRadio test update for baremetal support

0xc0170 committed:
Merge pull request #13979 from taunomagnusson/master

Support for ST STM32 NUCLEO-G431KB Development Board

0xc0170 committed:
Merge pull request #14213 from 0xc0170/dev_add_cortexm55

Add cortex-m55 to CMake support

0xc0170 committed:
Merge pull request #14190 from 0xc0170/fix_ns_cores

CMake cores: fix NS cores

0xc0170 committed:
Merge pull request #14204 from multiplemonomials/cmake-exe-suffix

Set executable suffix in CMake

0xc0170 committed:
Merge pull request #14228 from ARMmbed/rename_greentea_test_macro

CMake: rename greentea test macro

0xc0170 committed:
Merge pull request #14203 from kylejansen/sdpk1-usb

Adding USB HS Support to the SDP-K1

0xc0170 committed:
Merge pull request #14215 from flippy84/master

STM: Fix Hal_GetTick wrap around errors

0xc0170 committed:
Merge pull request #14199 from rwalton-arm/dev/rwalton-arm/refactor-st-mbed-targets

Refactor ST Mbed targets to be CMake buildsystem targets

0xc0170 committed:
Merge pull request #14031 from arduino/blockdevices_namespaces

BlockDevices: specify mbed namespace where needed

JeanMarcR committed:
STM32WL : ADDING CORTEX-M4 LINK OPTION

Add link option in file used for compilation.
Due to a bug in mbedtools environment.
